sunpy/coordinates/frames.py

```diff
@@ -587,6 +587,12 @@ def make_3d(self):
 
             d = np.fmin(d, dd) if self._spherical_screen['only_off_disk'] else dd
 
+        if np.all(np.isnan(d)) and np.any(np.isfinite(cos_alpha)):
+            warn_user("The conversion of these 2D helioprojective coordinates to 3D is all NaNs "
+                      "because off-disk coordinates need an additional assumption to be mapped to "
+                      "3D. Consider using the context manager "
```

Comment:
Maybe something like this would clarify this a little?

```suggestion
                      "because off-disk coordinates need an additional assumption to be mapped to calculate distance from the observer."
                      "Consider using the context manager "
```